code/crates/core-types/src/signing.rs

```diff
@@ -406,4 +471,74 @@ where
             })
         }
     }
+
+    /// Verify the round certificate against the given validator set.
+    ///
+    /// - For each signature in the certificate:
+    ///   - Reconstruct the signed vote and verify its signature.
+    /// - Check that the required voting power has signed the certificate:
+    ///   - If `Precommit`, ensure that 2/3+ of the voting power is represented.
+    ///   - If `Skip`, ensure that 1/3+ of the voting power is represented.
+    ///  
+    /// Returns a [`CertificateError`] if any verification step fails.
+    fn verify_round_certificate(
```

Comment:
Could you add some tests in `crates/test/tests/unit/certificates/round.rs` (new file). There are some for other certificates in the same dir, e.g. `polka.rs`.